drivers/npmx/npmx_shell.c

```diff
@@ -2706,6 +2727,289 @@ static int cmd_vbusin_status_connected_get(const struct shell *shell, size_t arg
 	return 0;
 }
 
+static int ship_config_get(const struct shell *shell, ship_config_type_t config_type)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	npmx_ship_config_t config;
+	npmx_ship_t *ship_instance = npmx_ship_get(npmx_instance, 0);
+	npmx_error_t err_code = npmx_ship_config_get(ship_instance, &config);
+
+	if (check_error_code(shell, err_code)) {
+		uint32_t config_val = 0;
+
+		switch (config_type) {
+		case SHIP_CONFIG_TYPE_TIME:
+			npmx_ship_time_convert_to_ms(config.time, &config_val);
+			break;
+
+		case SHIP_CONFIG_TYPE_INV_POLARITY:
+			config_val = (uint32_t)config.inverted_polarity;
+			break;
+		}
+
+		shell_print(shell, "Value: %d.", (int)config_val);
+	} else {
+		shell_error(shell, "Error: unable to read config.");
+	}
+
+	return 0;
+}
+
+static int ship_config_set(const struct shell *shell, size_t argc, char **argv,
+			   ship_config_type_t config_type)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	if (argc < 2) {
+		shell_error(shell, "Error: missing config value.");
+		return 0;
+	}
+
+	int err = 0;
+	uint32_t config_val = shell_strtoul(argv[1], 0, &err);
+
+	if (err != 0) {
+		shell_error(shell, "Error: config value must be an integer.");
+		return 0;
+	}
+
+	npmx_ship_config_t config;
+	npmx_ship_t *ship_instance = npmx_ship_get(npmx_instance, 0);
+
+	npmx_error_t err_code = npmx_ship_config_get(ship_instance, &config);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to read config.");
+		return 0;
+	}
+
+	switch (config_type) {
+	case SHIP_CONFIG_TYPE_TIME:
+		npmx_ship_time_t time = npmx_ship_time_convert(config_val);
+		if (time == NPMX_SHIP_TIME_INVALID) {
+			shell_error(shell, "Error: wrong time value.");
+			return 0;
+		}
+		config.time = time;
+		break;
+
+	case SHIP_CONFIG_TYPE_INV_POLARITY:
+		config.inverted_polarity = !!config_val;
+		break;
+	}
+
+	err_code = npmx_ship_config_set(ship_instance, &config);
+
+	if (check_error_code(shell, err_code)) {
+		shell_print(shell, "Success: %d.", (int)config_val);
```

Comment:
Same as in `ship_reset_config_set`.

Reply:
Then it should be like this:
https://github.com/NordicSemiconductor/npmx-zephyr/pull/59#discussion_r1330189876